### PR TITLE
TFAC-759 Creating Private Preview Page

### PR DESCRIPTION
**Pull request by jedtan**

### CODE REVIEW

**Comment by jedtan** [comment]
This is currently what the home page looks like in the case the page is a preview: 
image

I like the snackbar more than the little div at the top of the page since you'll always be able to see the reminder to not share the page.

Also for the URL scheme I just appended the cause ID to the end of the URL to make it a little harder to guess too.

File: gatsby-node.esm.js [hunk 1]
```diff
@@ -308,9 +317,18 @@ exports.createPages = async ({ actions, graphql }) => {
             },
           })
         })
+      } else {
+        // No need to create referrer URLs 
+        createPage({
+          path: `${path}/${data.causeId}/`,
```
Comment (kmjennison) [inline gatsby-node.esm.js:323]: For more external readability, let's do `${path}/${node.path}/preview/`. This wasn't in the spec, sorry!

File: gatsby-node.esm.js [hunk 2]
```diff
@@ -308,9 +317,18 @@ exports.createPages = async ({ actions, graphql }) => {
...
+          component: causeLaunchData.preview ? HomePageWrapper : NotFoundPage,
+          context: {
+            data: pivotedData,
+          },
+        })
       }
-    )
-  }
+    }
+  )
```
Comment (kmjennison) [inline gatsby-node.esm.js:331]: I think this is missing this acceptance criterion: "If vertical has a preview page enabled and the vertical has already launched, it does not render content and redirects to the real landing page for the vertical. Client-side redirect is OK." Currently, the preview page would 404, right? The goal with the redirect is to provide content to any people who are still using the preview link for whatever reason.

BTW, the [Gatsby S3 plugin](https://www.gatsbyjs.com/plugins/gatsby-plugin-s3/) that we use might help with redirects. But if it's not immediately easy to work in this case, just go with a client-side redirect.

File: src/components/V4HomePage.js [hunk 3]
```diff
@@ -92,9 +97,15 @@ const HomepageWrapper = ({
...
+            {referrer || isPreviewPage ? (
+              <meta name="robots" content="noindex" />
+            ) : null}
           </Helmet>
-          <Landing landingData={landing} causeId={causeId} />
+          <Landing
+            landingData={landing}
+            causeId={causeId}
+            previewMessage={isPreviewPage}
+          />
```
Comment (kmjennison) [inline src/components/V4HomePage.js:108]: For UI testing, could you add a Chromatic variant with the preview enabled (probably just for the `Landing` component)?

**Review by kmjennison — CHANGES_REQUESTED** [review]
Nice work! A few requested changes.

File: gatsby-node.esm.js (hunk 2 above)
Comment (jedtan) [reply gatsby-node.esm.js:331]: Ohh, yes I just missed that.

**Review by kmjennison — APPROVED** [review]
LGTM! 🦊

**Pull request merged**